src/ApiService/ApiService/onefuzzlib/VmssOperations.cs

```diff
@@ -85,12 +85,17 @@ public class VmssOperations : IVmssOperations {
     public async Async.Task<OneFuzzResultVoid> ResizeVmss(Guid name, long capacity) {
         var canUpdate = await CheckCanUpdate(name);
         if (canUpdate.IsOk) {
-            _log.Info($"updating VM count {name:Tag:VmssName} - {capacity:Tag:Count}");
             var scalesetResource = GetVmssResource(name);
             var patch = new VirtualMachineScaleSetPatch();
             patch.Sku.Capacity = capacity;
-            _ = await scalesetResource.UpdateAsync(WaitUntil.Started, patch);
-            return OneFuzzResultVoid.Ok;
+            try {
+                _log.Info($"updating VM count {name:Tag:VmssName} - {capacity:Tag:Count}");
+                _ = await scalesetResource.UpdateAsync(WaitUntil.Started, patch);
+                return OneFuzzResultVoid.Ok;
+            } catch (RequestFailedException ex) {
```

Comment:
Do we want to catch all exceptions or specifically the model exceptions?

We could follow a model similar to here:
https://github.com/microsoft/onefuzz/blob/f20c482052178ea20d9adf13d063bb979e900607/src/ApiService/ApiService/onefuzzlib/NodeOperations.cs#L400-L402

And maybe add ex.Message.contains("has reached its limit of 10 models that may be referenced by one or more VMs belonging to the Virtual Machine Scale Set. Upgrade the VMs to the latest model of the Virtual Machine Scale Set before trying again."

Reply:
Or even better, factor out that logic into a separate function so it's repeatable in the rest of the file! 

`catch (RequestFailedException ex) when (IsModelException(ex) {`

then

```csharp
static bool IsModelException(RequestFailedException ex) {
    return ex.Status == 400 
        && ex.ErrorCode == "BadRequest" 
        && ex.Message.Contains("has reached its limit of 10 models that may be referenced by one or more VMs belonging to the Virtual Machine Scale Set. Upgrade the VMs to the latest model of the Virtual Machine Scale Set before trying again.");
}

Reply:
I don't neccessarily see an issue with catching all exceptions considering how much we monitor them. If any of these calls fail, then we get into the pattern of not processing scalesets fully and having issues like node's state hanging.

Reply:
Fair enough, we're logging the exception, so we know at least it won't go unnoticed.